golem/managers/proposal/plugins/negotiating/add_mid_agreement_payments.py

```diff
@@ -0,0 +1,95 @@
+        max_demand_payment_timeout: timedelta = timedelta(hours=24),
+    ) -> None:
+        self._min_demand_debit_note_interval: int = int(
+            min_demand_debit_note_interval.total_seconds()
+        )
+        self._max_demand_debit_note_interval: int = int(
+            max_demand_debit_note_interval.total_seconds()
+        )
+        self._min_demand_payment_timeout: int = int(min_demand_payment_timeout.total_seconds())
+        self._max_demand_payment_timeout: int = int(max_demand_payment_timeout.total_seconds())
+
+    async def __call__(self, demand_data: DemandData, proposal_data: ProposalData) -> None:
+        offer_debit_note_interval = proposal_data.properties.get(
+            "golem.com.scheme.payu.debit-note.interval-sec?"
+        )
+        offer_payment_timeout = proposal_data.properties.get(
+            "golem.com.scheme.payu.payment-timeout-sec?"
+        )
+        if offer_debit_note_interval is None or offer_payment_timeout is None:
+            raise RejectProposal("Offer doesn't support mid agreement payments")
+
+        demand_debit_note_interval = demand_data.properties.get(
+            "golem.com.scheme.payu.debit-note.interval-sec?"
+        )
+        demand_payment_timeout = demand_data.properties.get(
+            "golem.com.scheme.payu.payment-timeout-sec?"
+        )
+
+        if (
+            offer_debit_note_interval == demand_debit_note_interval
+            and offer_payment_timeout == demand_payment_timeout
+        ):
+            logger.debug(
+                "Mid agreement properties negotiation done with debit note interval: "
+                f"{offer_debit_note_interval} and  payment timeout: {offer_payment_timeout}"
+            )
+            return
+
+        demand_data.properties[
+            "golem.com.scheme.payu.debit-note.interval-sec?"
+        ] = self._calculate_new_value_proposition(
+            offer_debit_note_interval,
+            demand_debit_note_interval,
+            self._min_demand_debit_note_interval,
+            self._max_demand_debit_note_interval,
+        )
+        demand_data.properties[
+            "golem.com.scheme.payu.payment-timeout-sec?"
+        ] = self._calculate_new_value_proposition(
+            offer_payment_timeout,
+            demand_payment_timeout,
+            self._min_demand_payment_timeout,
+            self._max_demand_payment_timeout,
+        )
+        logger.debug(
+            "Ongoing mid agreement properties negotiation with debit note interval: "
+            f"{demand_data.properties['golem.com.scheme.payu.debit-note.interval-sec?']}"
+            " and payment timeout: "
+            f"{demand_data.properties['golem.com.scheme.payu.payment-timeout-sec?']}"
+        )
+        return
+
+    @staticmethod
+    def _calculate_new_value_proposition(
+        offer: int, previous_demand: Optional[int], min_demand: int, max_demand: int
+    ):
+        if offer == previous_demand:
```

Comment:
```suggestion
        if previous_demand and offer >= previous_demand:
```

shouldn't we be satisfied with a value that's larger than our demand?

Reply:
This way user wont be able to limit `max`.

So it depend if we want to allow it or not.